### PR TITLE
Register plugin routes before the Medusa core routes

### PR DESCRIPTION
**Pull request by thetutlage**
Load plugin routes before the Medusa core routes, since it will allow the plugin to define routes with higher priority than Medusa. Here are couple of examples.

- Plugin registers a route called "/products/active"
- Medusa registers a route called "/products/:id"

Now, if Medusa routes gets registered first, then the `/products/active` route will never be resolved, because it will be handled by the `/products/:id` route.

### CODE REVIEW

**Review by sradevski — APPROVED** [review]
We need to revisit our entire routing TBH as it is a bit of a mess, but this works for now

File: packages/medusa/src/loaders/api.ts [hunk 1]
```diff
@@ -28,6 +28,28 @@ export default async ({ app, container, plugins }: Options) => {
...
+   */
+  await Promise.all(
+    plugins.map(async (pluginDetails) => {
+      return new RoutesLoader({
+        app: app,
+        configModule,
+        rootDir: path.join(pluginDetails.resolve, "api"),
+      }).load()
+    })
+  )
```
Comment (riqwan) [inline packages/medusa/src/loaders/api.ts:51]: sugg: to wrap this baby up with a user friendly error

**Review by riqwan — APPROVED** [review]
LGTM!

File: packages/medusa/src/loaders/api.ts (hunk 1 above)
Comment (thetutlage) [reply packages/medusa/src/loaders/api.ts:51]: Agree 100%.

But usually we have these sorts of loaders in many places and most of them are not wrapped in `try/catch` and then showing a user friendly error.

Maybe, we have to find all of them and settle on a pattern to use.

**Pull request merged**